dll/win32/shell32/dialogs/drive.cpp

```diff
@@ -234,6 +248,41 @@ SH_ShowDriveProperties(WCHAR *pwszDrive, IDataObject *pDataObj)
     return ret != -1;
 }
 
+BOOL
+SH_ShowDriveProperties(WCHAR *pwszDrive, IDataObject *pDataObj)
+{
+    HRESULT hr = SHStrDupW(pwszDrive, &pwszDrive);
+    if (FAILED_UNEXPECTEDLY(hr))
+        return FALSE;
+
+    // Prepare data for thread
+    DRIVE_PROP_DATA *pData = (DRIVE_PROP_DATA *)SHAlloc(sizeof(*pData));
+    if (!pData)
+    {
+        SHFree(pwszDrive);
+        return FALSE;
+    }
+    pData->pwszDrive = pwszDrive;
+
+    // Marshall IDataObject to IStream
+    hr = CoMarshalInterThreadInterfaceInStream(IID_IDataObject, pDataObj, &pData->pStream);
+    if (FAILED_UNEXPECTEDLY(hr))
+    {
+        SHFree(pwszDrive);
+        SHFree(pData);
+        return FALSE;
+    }
+
+    // Run a property sheet in another thread
+    if (SHCreateThread(ShowDrivePropThreadProc, pData, CTF_COINIT, NULL))
+        return TRUE; // Success
+
+    pData->pStream->Release();
+    SHFree(pwszDrive);
+    SHFree(pData);
```

Comment:
```suggestion
    SHFree(pData);
    SHFree(pwszDrive);
```
destruction in reverse order of allocation.